lib/msal-core/src/messaging/MessageListener.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { WindowType, MessageHelper } from "./MessageHelper";
+import { WindowUtils } from "./../utils/WindowUtils";
+import { MessageType, PAYLOAD } from "./MessageHelper";
+import { MessageCache } from "./MessageCache";
+import { MessageDispatcher } from "./MessageDispatcher";
+
+import { Logger } from "./../Logger";
+
+export class MessageListener {
+
+    private messageCache: MessageCache;
+    private logger: Logger;
+
+    /**
+     * initialize the message listener, and register the callback to process messages
+     */
+    constructor(messageCache: MessageCache, logger: Logger) {
+        this.messageCache = messageCache;
+        this.logger = logger;
+        // listen to all incoming messages
+        window.addEventListener("message", this.receiveMessage, false);
+    }
+
+    /**
+     * Parses the messages receieved
+     * This will be a unique handler per message, we will allow only one active request at a time
+     * @param event
+     */
+    private receiveMessage(event: any) {
```

Comment:
Is there not a specific type you can use?

Reply:
I tried MessageEvent but it did not seem to work. I did not find anything appropriate except `Object` or `any`.

Reply:
Hmm why didnt `MessageEvent` work?

Reply:
Interested to know why message event didn't work, seems like we lose some type checking there...

Reply:
Message dispatching with event.source which should be a `WindowProxy` is not type matching. I will dig into this later with a separate PR.